python/tests/expressions/test_expressions_base.py

```diff
@@ -257,3 +317,111 @@ def test_bound_reference(table_schema_simple, foo_struct):
     assert bound_ref1.eval(foo_struct) == "foovalue"
     assert bound_ref2.eval(foo_struct) == 123
     assert bound_ref3.eval(foo_struct) == True
+
+
+def test_boolean_expression_visitor():
+    """Test post-order traversal of boolean expression visit method"""
+    expr = base.And(
+        base.Or(base.Not(TestExpressionA()), base.Not(TestExpressionB()), TestExpressionA(), TestExpressionB()),
+        base.Not(TestExpressionA()),
+        TestExpressionB(),
+    )
+    visitor = TestBooleanExpressionVisitor()
+    result = base.visit(expr, visitor=visitor)
+    assert result == [
+        "TestExpressionA",
+        "NOT",
+        "TestExpressionB",
+        "NOT",
+        "OR",
+        "TestExpressionA",
+        "OR",
+        "TestExpressionB",
+        "OR",
+        "TestExpressionA",
+        "NOT",
+        "AND",
+        "TestExpressionB",
+        "AND",
+    ]
+
+
+def test_or_with_always_true():
```

Comment:
This doesn't test the visitor, it is testing `Or.__new__` that simplifies the expression as it is created. I think you can remove these and/or/false/true tests and just go with the one above for now.

Reply:
Removed 👍